libraries/dagster-hex/dagster_hex/resources.py

```diff
@@ -25,30 +28,28 @@
 )
 
 
-class HexResource:
-    """
-    This class exposes methods on top of the Hex REST API.
+class HexResource(dg.ConfigurableResource):
+    api_key: str = Field(
+        ...,
+        description="Hex API Key. You can find this on the Hex settings page",
+    )
 
-    Args:
-        api_key (str): Hex API Token to use for authentication
-        base_url (str): Base URL for the API
-        request_max_retries (int): Number of times to retry a failed request
-        request_retry_delay (int): Time, in seconds, to wait between retries
-    """
+    base_url: str = Field(
+        default=HEX_API_BASE,
+        description="Hex Base URL for API requests.",
+    )
 
-    def __init__(
-        self,
-        api_key: str,
-        base_url: str = HEX_API_BASE,
-        log: logging.Logger = get_dagster_logger(),
-        request_max_retries: int = 3,
-        request_retry_delay: float = 0.25,
-    ):
-        self._log = log
-        self._api_key = api_key
-        self._request_max_retries = request_max_retries
-        self._request_retry_delay = request_retry_delay
-        self.api_base_url = base_url
+    request_max_retries: int = Field(
+        default=3,
+        description="The maximum times requests to the Hex API should be retried before failing.",
+    )
+
+    request_retry_delay: float = Field(
+        default=0.25,
+        description="Time (in seconds) to wait between each request retry.",
+    )
+
+    _log: Logger = PrivateAttr(dg.get_dagster_logger())
```

Comment:
Good call. Do you know if the following would work?
```
    log: Logger = Field(
        default=dg.get_dagster_logger(),
        description="Optional logger to be used from within the HexResource (default: `get_dagster_logger()`",
    )
```
Or would we need to do something along the lines of:
```
    log: Logger = Field(
        description="Optional logger to be used from within the HexResource (default: `get_dagster_logger()`",
    )


    @property
    @cached_method
    def _log(self) -> logging.Logger:
        return self.log if self.log else dg.get_dagster_logger()
```

Reply:
I think the second example works well if we want to allow custom loggers

Reply:
Ah shoot, it looks like it might not work actually.
```
E  dagster._core.errors.DagsterInvalidPythonicConfigDefinitionError:
E  Error defining Dagster config class <class 'dagster_hex.resources.HexResource'> on field 'log'.
E  Unable to resolve config type <class 'logging.Logger'> to a supported Dagster config type.
E
E
E  This config type can be a:
E      - Python primitive type
E          - int, float, bool, str, list
E      - A Python Dict or List type containing other valid types
E      - Custom data classes extending dagster.Config
E      - A Pydantic discriminated union type (https://docs.pydantic.dev/usage/types/#discriminated-unions-aka-tagged-unions)
E
E
E  If this config type represents a resource dependency, its annotation must either:
E      - Extend dagster.ConfigurableResource, dagster.ConfigurableIOManager, or
E      - Be wrapped in a ResourceDependency annotation, e.g. ResourceDependency[Logger]
```

Reply:
Or maybe the suggestion of wrapping it in a `ResourceDependency` works! 🤔 
```
    log: dg.ResourceDependency[logging.Logger] = Field(
        default_factory=dg.get_dagster_logger,
        description="Optional logger to be used from within the HexResource (default: `get_dagster_logger()`",
    )
```
Tests pass after making that change.

Reply:
Great. It looks like the default logger is not instantiated though. 

Also, do you think we could use the cached method like in your second example?   

```python
@property
@cached_method
def _log(self) -> logging.Logger:
    return self.log if self.log else dg.get_dagster_logger()
```

Reply:
Updated to use `cached_method`, thank you!